plenum/test/primary_selection/helper.py

```diff
@@ -0,0 +1,42 @@
+from plenum.server.pool_manager import RegistryPoolManager, TxnPoolManager
+from plenum.test.test_node import checkProtocolInstanceSetup
+
+
+def check_rank_consistent_across_each_node(nodes):
+    assert nodes
+    node_ranks = {}
+    name_by_ranks = {}
+    for node in nodes:
+        node_ranks[node.poolManager.id] = node.rank
+        name_by_ranks[node.rank] = node.name
+
+    for node in nodes:
+        for other_node in nodes:
+            if node != other_node:
+                oid = other_node.poolManager.id
+                assert node.poolManager.get_rank_of(oid) ==  node_ranks[oid]
+                ork = node_ranks[oid]
+                assert node.poolManager.get_name_by_rank(ork) ==  name_by_ranks[ork]
+    order = []
+    for node in nodes:
+        if isinstance(node.poolManager, RegistryPoolManager):
```

Comment:
I think we pass nodes with the implementation of pool manager all the time, don't we?
We need to have separate explicit tests for each of pool ledger implementations.